scripts/globals/pets/avatar.lua

```diff
@@ -0,0 +1,385 @@
+            buffTarget = master
+        end
+
+        local distance  = pet:checkDistance(master) -- starts as distance to master, updated to be distance to the posTarget
+        local hpp       = 0
+        local alliance  = master:getAlliance()
+        local party     = master:getParty()
+        local petLvl    = pet:getMainLvl()
+
+        for _, member in pairs(alliance) do
+            -- tiered chance of healing based on how low they are
+            local tempHPP = member:getHPP()
+            if
+                pet:checkDistance(member) < 20 and
+                (hpp == 0 or
+                tempHPP < hpp) and
+                ((tempHPP < 75 and
+                math.random(100) < 50) or
+                (tempHPP < 50 and
+                math.random(100) < 50) or
+                (tempHPP < 25 and
+                math.random(100) < 50))
+            then
+                hpp = tempHPP
+                cureTarget = member
+            end
+        end
+
+        if cureTarget then
+            -- curaga can be used on anyone in the alliance and isn't determined by how many need a heal (assumed to be based on skill over cap)
+            if
+                petLvl >= 16 and
+                math.random(100) <= 2 * xi.summon.getSummoningSkillOverCap(pet)
+            then
+                -- curaga
+                if petLvl >= 71 then
+                    spellID = xi.magic.spell.CURAGA_IV
+                elseif petLvl >= 51 then
+                    spellID = xi.magic.spell.CURAGA_III
+                elseif petLvl >= 31 then
+                    spellID = xi.magic.spell.CURAGA_II
+                else
+                    spellID = xi.magic.spell.CURAGA
+                end
+            else
+                if petLvl >= 61 then
+                    spellID = xi.magic.spell.CURE_V
+                elseif petLvl >= 41 then
+                    spellID = xi.magic.spell.CURE_IV
+                elseif petLvl >= 21 then
+                    spellID = xi.magic.spell.CURE_III
+                elseif petLvl >= 11 then
+                    spellID = xi.magic.spell.CURE_II
+                else
+                    spellID = xi.magic.spell.CURE
+                end
+            end
+
+            return spellID, cureTarget
+        else
+            local hasteTier = { 0, 0 }
+            if petLvl >= 40 then
+                hasteTier = { 1, 15 }
+            end
+
+            local regenTier = { 0, 0 }
+            if petLvl >= 21 then
+                regenTier = { 1, 1 }
+            end
+
+            local shellTier = { 0, 0 }
+            if petLvl >= 68 then
+                shellTier = { 4, 2617 }
+            elseif petLvl >= 52 then
+                shellTier = { 3, 2188 }
+            elseif petLvl >= 37 then
+                shellTier = { 2, 1641 }
+            elseif petLvl >= 17 then
+                shellTier = { 1, 1055 }
+            end
+
+            local protectTier = { 0, 0 }
```

Comment:
This block looks suspiciously like it can be written as a table+driver

Reply:
I couldn't for the life of me think of a clean way to make this a table

Reply:
Done. Hopefully you like it. I'm pretty happy of the results